core/vm/evm.go

```diff
@@ -42,11 +42,32 @@ type (
 	GetHashFunc func(uint64) common.Hash
 )
 
+// ActivePrecompiles returns the addresses of the precompiles enabled with the current
+// configuration
+func (evm *EVM) ActivePrecompiles() []common.Address {
+	var precompiles map[common.Address]PrecompiledContract
+	switch {
+	//case evm.chainRules.IsYoloV2:
+	//	precompiles = PrecompiledContractsYoloV1
+	case evm.chainRules.IsIstanbul:
+		precompiles = PrecompiledContractsIstanbul
+	case evm.chainRules.IsByzantium:
+		precompiles = PrecompiledContractsByzantium
+	default:
+		precompiles = PrecompiledContractsHomestead
+	}
+	var addresses []common.Address
+	for k, _ := range precompiles {
+		addresses = append(addresses, k)
+	}
+	return addresses
+}
```

Comment:
If we need a list of addresses for every transaction, it might make sense to have declared globals the same way we have for the maps themselves. Otherwise we're redoing the same useless work of flattening maps into addresses over and over and over and over again for all 600M transactions.